### PR TITLE
test(e2e): per-metric collaboration specs for the remaining m365 metrics (#1413)

### LINKED ISSUE #1441: Metric smoke: Collaboration metrics
## Collaboration — Metric smoke

Gold view `insight.collab_bullet_rows` · flows …0005 (Team) / …0012 (IC) · grain member · team. Source connectors: m365, slack, zoom.

> **Happy-path *smoke* only.** Seed one clean, deterministic bronze dataset → assert the served values match the golden CSV. Negative / auth / empty / malformed / edge cases are **out of scope** (separate negative-test track). Fixture: `collab_smoke`. Runs in the rig (#1422).

## Target metrics (20)

- **Active Days** — `m365_active_days` · ✅ real
- **Emails Read** — `m365_emails_read` · ✅ real
- **Emails Received** — `m365_emails_received` · ✅ real
- **Emails Sent** — `m365_emails_sent` · ✅ real
- **Files Engaged** — `m365_files_engaged` · ✅ real
- **Files Shared (External)** — `m365_files_shared_external` · ✅ real
- **Files Shared (Internal)** — `m365_files_shared_internal` · ✅ real
- **Teams Chats** — `m365_teams_chats` · ✅ real
- **Meeting-Free Days** — `meeting_free` · ✅ real
- **Meeting Hours** — `meeting_hours` · ✅ real
- **Meetings Attended** — `meetings_count` · ✅ real
- **Active Days** — `slack_active_days` · ✅ real
- **Channel Posts** — `slack_channel_posts` · ✅ real
- **Messages Sent** — `slack_messages_sent` · ✅ real
- **Teams Meeting Hours** — `teams_meeting_hours` · ✅ real
- **Teams Meetings Attended** — `teams_meetings` · ✅ real
- **Zoom Meeting Hours** — `zoom_meeting_hours` · ✅ real
- **Zoom Meetings Attended** — `zoom_meetings` · ✅ real
- **DM Ratio** — `slack_dm_ratio` · 🔁 reconstructed
- **Messages per Active Day** — `slack_msgs_per_active_day` · 🔁 reconstructed

*Legend — ✅ real (assert the gold value) · 🔁 reconstructed in `query_ref` (Σnum/Σden — assert the API value) · ⛔ locked stub / ComingSoon (assert NULL).*

## Pipeline
bronze (m365, slack, zoom) → silver `class_*` (read FINAL) → gold `insight.collab_bullet_rows` → API `POST /v1/metrics/{id}/query`.

## Notes
Meeting metrics read `class_collab_meeting_activity FINAL` directly. **Absorbs #1413** — its m365 seed work feeds this smoke (m365 ≈ 12 of 20 keys).

## Acceptance
- [ ] `collab_smoke` seeds clean bronze → asserts every target metric's served value vs golden CSV (member · team).
- [ ] Reconstructed ratios asserted at the API layer; locked stubs asserted NULL.
- [ ] Runs green in the e2e rig (#1422) in CI.

Parent: #1438 (Metric smokes) · Epic #1404 · P0

### PR DESCRIPTION
**Pull request by ktursunov**
## Summary

Completes the per-metric E2E coverage of the m365 **Collaboration** bullet. The
email metrics landed in #1472; this PR adds the remaining 10 metrics, each driving
the full `bronze → dbt silver → gold view → analytics-api` path and asserting on
the IC bullet (`…0012`) over a one-team (Engineering) cohort.

| stream (bronze) | metrics |
|---|---|
| **Teams** (`teams_activity`) | `m365_teams_chats`, `meetings_count`, `teams_meetings`, `meeting_hours`, `teams_meeting_hours`, `meeting_free` |
| **Files** (`onedrive_activity` + `sharepoint_activity`) | `m365_files_engaged`, `m365_files_shared_internal`, `m365_files_shared_external` |
| **multi-stream** (email + onedrive + teams) | `m365_active_days` |

Each spec:
- asserts **only** the target `metric_key`'s `value` / `median` / `range_*` (value = the
  member, median/range = the team cohort), per the skill convention;
- seeds a **verbatim-duplicate** alice row so RMT/`union_by_tag` dedup is exercised on
  each new silver class (`class_collab_chat_activity`, `class_collab_meeting_activity`,
  `class_collab_document_activity`) — the value must dedup, not double.

Also adds the `teams`/`onedrive`/`sharepoint` bronze schemas + `$ref` templates.

## Testing

```
cd src/ingestion/tests/e2e && ./e2e.sh test
======================== 45 passed in 117.77s ========================
```

(10 new specs + the email/zulip specs + meta framework tests.) Goldens were
confirmed twice — once before and once after adding the dedup rows (unchanged,
proving dedup).

## Scope / follow-up

Covers all 13 m365 collaboration bullet metric_keys (3 email in #1472 + these 10).
Still deferred (separate concerns, can follow up): the `collab_team_cohort` spec
(Team bullet `…0005`, multi-team) and the `collab_smoke` integration spec.

## Summary by CodeRabbit

* **New Features**
  * Expanded end-to-end coverage for collaboration metrics, including active days, meeting-free, meeting hours/counts, Teams chats, Teams meetings/hours, and internal/external file engagement & sharing.
* **Bug Fixes**
  * Improved end-to-end determinism by truncating additional Zoom staging tables at session start.
  * Strengthened validation that duplicate activity rows are deduplicated and not double-counted.
* **Documentation / Chores**
  * Added reusable fixture templates and JSON schemas for M365 and Zoom activity, plus expanded Zoom bronze placeholders.
  * Refined end-to-end metric test guidance and updated collaboration test documentation text.

### CODE REVIEW

File: src/ingestion/tests/e2e/specs/collab_teams_meetings.test.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      - assert: "status == 200"
+      - in: collaboration
+        assert: "result.status == 'ok'"
+      - in: collaboration
+        find: { metric_key: teams_meetings }
+        equal: { value: 4, median: 2, range_min: 1, range_max: 4 }
+
+  # SAME fixture: meetings_count INCLUDES the Zoom meeting → alice = 4 Teams + 1 Zoom = 5
+  #   team {5,2,1} → median 2, range [1,5]. Proves the Zoom row landed.
+  - name: teams_meetings — Zoom counts in cross-source meetings_count (value 5)
```
Comment (mitasovr) [inline src/ingestion/tests/e2e/specs/collab_teams_meetings.test.yaml]: technically you can check both cases in one request

Comment (ktursunov) [reply src/ingestion/tests/e2e/specs/collab_teams_meetings.test.yaml]: Deliberate additional check

Comment (ktursunov) [reply src/ingestion/tests/e2e/specs/collab_teams_meetings.test.yaml]: But it might be too much for this metric

**Review by mitasovr — APPROVED** [review]
